### DISCUSSION TITLE
PDM doesn't work after upgrading python

### DISCUSSION THREAD

**Discussion by ShaiAvr** [pdm-project/pdm#3050] [Q&A]
I use [pyenv on Windows](https://github.com/pyenv-win/pyenv-win) to manage my Python installations and I got a problem after upgrading my Python version. I had version `3.11.5` installed and I upgraded to `3.12.4` and deleted the old Python version. The problem is that now any time I open the terminal (Powershell), I get an error `No Python at '"D:\Portable_Programs\.pyenv\pyenv-win\versions\3.11.5\python.exe'`. Also, any command I try to run with `pdm` doesn't work with the same error (even simply running `pdm`).

How can I tell `pdm` to use the new Python version I installed? I thought to uninstall `pdm`, but to my surprise, I couldn't find any uninstallation instructions in the docs or online. It also doesn't seem like a good solution to reinstall `pdm` anytime I update Python.

Any help would be appreciated.

**Comment by frostming** [comment]
Your shell completion is broken, try re-install the completion script https://pdm-project.org/en/latest/#shell-completion

**Reply by ShaiAvr** [reply]
But I can't use `pdm`. Every command results in `No Python at '"D:\Portable_Programs\.pyenv\pyenv-win\versions\3.11.5\python.exe'`.

**Comment by pawamoy** [comment]
It could also be that you installed PDM with pipx or pip --user on 3.11. That left an executable somewhere, that calls Python 3.11 which doesn't exist anymore. The solution is to reinstall PDM with 3.12 and remove the old executable if you can find it.

Nothing specific to PDM, that's how Python works 😐 Next time you upgrade, try to remember migrating all user-installed tools to the new Python version before deleting the old one.

**Reply by ShaiAvr** [reply]
I didn't install with `pip` or `pipx`. I used the installation script.

**Reply by frostming** [reply]
each python script relies on an interpreter to work. Removing an existing interpreter puts you in a risk that some tools may be broken, just re-install them if necessary.

**Reply by ShaiAvr** [reply]
But how? As I said, I couldn't find any uninstallation instructions, and `pdm` doesn't work at the moment, so I can't use it.

**Reply by frostming** [reply]
The install script includes an option to remove:

```ps
(Invoke-WebRequest -Uri https://pdm-project.org/install-pdm.py -UseBasicParsing).Content | py - --remove
```

**Reply by ShaiAvr** [reply]
Where can I find the config file of `pdm`? I want to restore my config after reinstalling.

**Reply by frostming** [reply]
You don't need to, the uninstallation won't delete the config files
